cloud/gce.go

```diff
@@ -245,6 +261,44 @@ func (p *GCEProvider) Create(id string, attr CreateAttributes) (Instance, error)
 	}
 }
 
+// Remove stops the instance with the given ID and terminates it.
+func (p *GCEProvider) Remove(id string) (Instance, error) {
+	state := &multistep.BasicStateBag{}
+
+	c := &gceStopContext{
+		id:       id,
+		instChan: make(chan Instance),
+		errChan:  make(chan error),
+	}
+
+	runner := &multistep.BasicRunner{
+		Steps: []multistep.Step{
+			&gceStopMultistepWrapper{c: c, f: p.stepDeleteInstance},
+		},
+	}
+
+	go runner.Run(state)
+
+	select {
+	case inst := <-c.instChan:
+		return inst, nil
+	case err := <-c.errChan:
+		return Instance{}, err
+	}
+}
+
+func (p *GCEProvider) stepDeleteInstance(c *gceStopContext) multistep.StepAction {
+	op, err := p.client.Instances.Delete(p.projectID, p.ic.Zone.Name, c.instance.Name).Do()
+	if err != nil {
+		cbcontext.LoggerFromContext(c.ctx).WithField("err", err).Error("error deleting instance")
```

Comment:
Not sure if this needs to be here, since we are also "returning" the error (by sending it to `errChan`, which then causes `Remove` to return it. Presumably whatever calls `GCEProvider.Remove` would log the error?